src/GridTilemap/CharBlockCache/CharBlockCache.test.ts

```diff
@@ -260,6 +260,60 @@ describe("CharBlockCache", () => {
     expect(hasBlockingCharOnOldPos).toBe(true);
   });
 
+  it("should find all characters", () => {
+    const charMock1 = <any>{
+      ...createCharMock("player1"),
+      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
+      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
+    };
+    charBlockCache.addCharacter(charMock1);
+    expect(
+      charBlockCache.findCharactersAt(new Vector2(0, 1), "someLayer").size
+    ).toBe(1);
+  });
+
+  it("should find multiple characters on the same position", () => {
+    const charMock1 = <any>{
+      ...createCharMock("player1"),
+      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
+      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
+    };
+    charBlockCache.addCharacter(charMock1);
+    const charMock2 = <any>{
+      ...createCharMock("player2"),
+      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
+      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
+    };
+    charBlockCache.addCharacter(charMock2);
+    expect(
+      charBlockCache.findCharactersAt(new Vector2(0, 1), "someLayer").size
+    ).toBe(2);
+  });
+
+  it("should not find a character on a different layer", () => {
+    const charMock1 = <any>{
+      ...createCharMock("player1"),
+      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
+      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
+    };
+    charBlockCache.addCharacter(charMock1);
+    expect(
+      charBlockCache.findCharactersAt(new Vector2(0, 1), "someOtherLayer").size
+    ).toBe(0);
+  });
+
+  it ("should not find characters on a different position", () => {
+    const charMock1 = <any>{
+      ...createCharMock("player1"),
+      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
+      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
+    };
+    charBlockCache.addCharacter(charMock1);
+    expect(
+      charBlockCache.findCharactersAt(new Vector2(0, 2), "someLayer").size
+    ).toBe(0);
+  });
```

Comment:
You could merge these tests into a single one with several expectations.

Set up a couple of charMocks that have names that speak for themselves.

Something like:

```javascript
it("should find all characters", () => {
    const charMock1 = <any>{
      ...createCharMock("player1"),
      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
    };
    const charMock2 = <any>{
      ...createCharMock("player1"),
      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "someLayer" }),
      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "someLayer" }),
    };
    
    const charMockDifferentLayer = <any>{
      ...createCharMock("player1"),
      getTilePos: () => ({ position: { x: 0, y: 1 }, layer: "otherLayer" }),
      getNextTilePos: () => ({ position: { x: 0, y: 2 }, layer: "otherLayer" }),
    };
    
    const charMockDifferentPos = <any>{
      ...createCharMock("player1"),
      getTilePos: () => ({ position: { x: 1 y: 1 }, layer: "someLayer" }),
      getNextTilePos: () => ({ position: { x: 1, y: 2 }, layer: "someLayerLayer" }),
    };
    charBlockCache.addCharacter(charMock1);
    // ...
    charBlockCache.addCharacter(charMockDifferentPos);
    expect(
      charBlockCache.findCharactersAt(new Vector2(0, 1), "someLayer").size
    ).toBe(2);
     expect(
      charBlockCache.findCharactersAt(new Vector2(0, 1), "otherLayer").size
    ).toBe(1);
    expect(
      charBlockCache.findCharactersAt(new Vector2(1, 1), "someLayer").size
    ).toBe(1);
     expect(
      charBlockCache.findCharactersAt(new Vector2(7, 7), "someLayer").size
    ).toBe(0);
  });

```

I think this is a bit more concise and therefore more readable.

Reply:
Thanks. Yes I agree. In my opinion it depends a lot on the concrete case.